l2geth/rollup/sync_service.go

```diff
@@ -174,31 +183,27 @@ func (s *SyncService) ensureClient() error {
 	return nil
 }
 
-// Start initializes the service, connecting to Ethereum1 and starting the
-// subservices required for the operation of the SyncService.
-// txs through syncservice go to mempool.locals
-// txs through rpc go to mempool.remote
+// Start initializes the service
 func (s *SyncService) Start() error {
 	if !s.enable {
+		log.Info("Running without syncing enabled")
 		return nil
 	}
 	log.Info("Initializing Sync Service", "eth1-chainid", s.eth1ChainId)
 
-	// When a sequencer, be sure to sync to the tip of the ctc before allowing
-	// user transactions.
-	if !s.verifier {
-		err := s.syncTransactionsToTip()
-		if err != nil {
-			return fmt.Errorf("Cannot sync transactions to the tip: %w", err)
-		}
-		// TODO: This should also sync the enqueue'd transactions that have not
-		// been synced yet
-		s.setSyncStatus(false)
-	}
-
 	if s.verifier {
 		go s.VerifierLoop()
 	} else {
+		// The sequencer must sync the transactions to the tip and the
+		// pending queue transactions on start before setting sync status
+		// to false and opening up the RPC to accept transactions.
+		if err := s.syncTransactionsToTip(); err != nil {
+			return fmt.Errorf("Sequencer cannot sync transactions to tip: %w", err)
+		}
+		if err := s.syncQueueToTip(); err != nil {
+			log.Error("Sequencer cannot sync queue", "msg", err)
```

Comment:
Why do we log here but return an error above?

Reply:
Good catch, it makes more sense to also return an error there